assets/src/edit-story/components/panels/layer/layer.js

```diff
@@ -76,15 +80,20 @@ const LayerDescription = styled.div`
 
 function Layer({ element }) {
   const { LayerIcon, LayerContent } = getDefinitionForType(element.type);
+  const id = `layer-${element.id}`;
 
   const { isSelected, handleClick } = useLayerSelection(element);
 
   return (
-    <LayerButton isSelected={isSelected} onClick={handleClick}>
+    <LayerButton
+      isSelected={isSelected}
+      onClick={handleClick}
+      aria-labelledby={id}
```

Comment:
I also don't think this is strictly necessary, but it seems right. The layer description is the only textual content of the button anyway, but theoretically the icon could have a textual description, which would now be ignored.

Reply:
IMHO this is unnecessary. But ok by me if you feel strongly about it.

Reply:
Actually the icon now has the description of the layer type and the layer content only the specific instance (so e.g. the image filename). Much better since 2f08257.